src/util/address.rs

```diff
@@ -727,6 +727,36 @@ impl Address {
             (Network::Testnet, _) | (Network::Regtest, _) | (Network::Signet, _) => true
         }
     }
+
+    /// Returns true if the given pubkey is directly related to the address payload.
+    ///
+    /// This is determined by directly comparing the address payload with either the
+    /// hash of the given public key or the segwit redeem hash generated from the
+    /// given key. For taproot addresses, the supplied key is assumed to be tweaked
+    pub fn is_related_to_pubkey(&self, pubkey: &PublicKey) -> bool {
+        let pubkey_hash = pubkey.pubkey_hash();
+        let payload = self.payload_as_bytes();
+
+        (*pubkey_hash == *payload) || (pubkey.to_bytes()[1..] == *payload) || (*segwit_redeem_hash(&pubkey_hash) == *payload)
```

Comment:
This is a very cool trick but still ignores parity. Doesn't seem to be a security issue but @apoelstra could have some good idea. :) (Hope it's not annoying to mention you, I assume you're good at this because of your great talk about crypto gotchas. :))

Reply:
It's fine to @ me as long as you also include flattering statements ;). (Kidding!)

@Kixunil ignoring the parity here is actualyl exactly the right thing to do -- when converting a full key to an x-only key, the parity bit is simply dropped. During signing, the secret key is conditionally negated to make sure that everything still works even if the original key's parity doesn't match Taproot's implicit parity.

Reply:
Cool, thanks for explaining!